tools/generator/src/Logger.swift

```diff
@@ -10,33 +9,46 @@ protocol Logger {
 }
 
 /// A `TextOutputStream` that writes to standard error.
-private class StderrOutputStream: TextOutputStream {
+final class StderrOutputStream: TextOutputStream {
     func write(_ string: String) {
         fputs(string, stderr)
     }
 }
 
+/// A `TextOutputStream` that writes to standard out.
+final class StdoutOutputStream: TextOutputStream {
+    func write(_ string: String) {
+        fputs(string, stdout)
+    }
+}
+
 /// The logger that is used when not running tests.
-class DefaultLogger: Logger {
-    private var standardError = StderrOutputStream()
+final class DefaultLogger<E: TextOutputStream, O: TextOutputStream>: Logger {
```

Comment:
Does it need to be generic? Does `print()` not work with protocols?

Reply:
See the other thread on the swift compiler crash.